src/main/java/reposense/parser/CsvParser.java

```diff
@@ -87,6 +91,10 @@ public CsvParser(Path csvFilePath, int expectedHeaderSize) throws IOException {
         } catch (InvalidCsvException ice) {
             throw new IOException(ice.getMessage(), ice);
         }
+
+        if (results.isEmpty()) {
+            throw new IOException(String.format(MESSAGE_ZERO_VALID_CONFIGS, csvFilePath.getFileName()));
```

Comment:
Should we define a new error type for this kind of error?

Reply:
Should we use the `InvalidCsvException` instead?
The only issue is that we would need to change the `throws` clause of this `parse` method, which means several test cases need to be updated.

I'll go ahead and make this change.

Reply:
Can I know which test cases should be updated? Do remember that in one pinned issue, we are decided to use generic exception handling.

Reply:
This will affect:
-  8 test cases in `AuthorConfigParserTest`
- 4 test cases in `GroupConfigParserTest`
- 8 test cases in `RepoConfigParserTest`
- 1 test case in `ArgsParserTest`

Reply:
Yes, I think we can take this chance to update the relevant test to generic exception handling.

Reply:
@fzdy1914, is it ok if I update the style of **all test cases within the affected files** , even if some tests are not affected by this change?

For example, only 1 test case in `ArgsParserTest` is affected by this change.
Should I also update the other tests in `ArgsParserTest` to the new style?

Reply:
@fzdy1914, I have updated the `parse` method in `CsvParser` to throw `InvalidCsvException` when the csv files are malformed. 

Because of this change in the `throws` clause of `parse`, I have had to update `RepoSense.java` as well as the following test files:
- `ArgsParserTest`
- `ConfigSystemTest`
- `RepoConfigParserTest`
- `GroupConfigParserTest`
- `AuthorConfigParserTest`
- `RepoConfigurationTest`